src/site/antora/modules/ROOT/pages/manual/pattern-layout.adoc

```diff
@@ -0,0 +1,1719 @@
+
+`%throwable{short.localizedMessage}` outputs the localized message.
+
+`%throwable\{n}` outputs the first `n` lines of the stack trace.
+
+Specifying `%throwable\{none}` or `%throwable\{0}` suppresses output of the exception.
+
+Use `{filters(packages)}` where _packages_ is a list of package names to suppress matching stack frames from stack traces.
+
+Use `{suffix(pattern)}` to add the output of _pattern_ at the end of each stack frame.
+
+Use a `{separator(...)}` as the end-of-line string, e.g., `separator(\|)`.
+The default value is the `line.separator` system property, which is operating system dependent.
+
+[WARNING]
+====
+Exception converter is not garbage-free.
+====
+
+[#converter-exception-extended]
+==== Exception (Extended)
+
+The same as xref:#converter-exception[the `%throwable` conversion], but also includes class packaging information
+
+.link:../javadoc/log4j-core/org/apache/logging/log4j/core/pattern/ThrowablePatternConverter.html[`ThrowablePatternConverter`] specifier grammar
+[source,text]
+----
+xEx|xException|xThrowable
+  { "none"
+  | "full"
+  | depth
+  | "short"
+  | "short.className"
+  | "short.fileName"
+  | "short.lineNumber"
+  | "short.methodName"
+  | "short.message"
+  | "short.localizedMessage"
+  }
+  {filters(package,package,...)}
+  {suffix(pattern)}
+  {separator(separator)}
+----
+
+Different from xref:#converter-exception[the `%throwable` conversion], at the end of each stack element of the exception, a string containing the name of the JAR file that contains the class or the directory the class is located in and the `Implementation-Version` as found in that JAR's manifest will be added.
+If the information is uncertain, then the class packaging data will be preceded by a `~` (tilde) character.
+
+[#converter-file]
+==== File
+
+Outputs the file name where the logging request was issued
+
+.link:../javadoc/log4j-core/org/apache/logging/log4j/core/pattern/FileLocationPatternConverter.html[`FileLocationPatternConverter`] specifier grammar
+[source,text]
+----
+F
+file
+----
+
+[WARNING]
+====
+Capturing the source location information to generate the file name of the caller is an expensive operation, and is not garbage-free.
+See xref:manual/layouts.adoc#LocationInformation[this section of the layouts page] for details.
+====
+
+[#converter-fqcn]
+==== FQCN
+
+Outputs the fully qualified class name of the logger
+
+.link:../javadoc/log4j-core/org/apache/logging/log4j/core/pattern/LoggerFqcnPatternConverter.html[`LoggerFqcnPatternConverter`] specifier grammar
+[source,text]
+----
+fqcn
+----
+
+[#converter-highlight]
+==== Highlight
+
+Adds ANSI colors to the result of the enclosed pattern based on the current event's xref:manual/customloglevels.adoc[logging level].
+Windows users should refer to xref:#jansi[].
+
+.link:../javadoc/log4j-core/org/apache/logging/log4j/core/pattern/HighlightConverter.html[`HighlightConverter`] specifier grammar
+[source,text]
+----
+highlight{pattern}{style}
+----
```

Comment:
The `style` attribute is by far the most complex one.
Its syntax is a comma-separated list of specifiers in the form:

- `DISABLEANSI`, `NOCONSOLENOANSI`: we might want to mention that these specifiers are accepted for BC, but redirect users to the `disableAnsi` and `noConsoleNoAnsi` configuration attributes of `PatternLayout`.
- `STYLE=(DEFAULT|LOGBACK)`: is replaced by a predefined map level->color specifier. This **should** appear first in the list.
- `<level_name>=<color_specifier>`: the `<color_specifier>` part is again a complex structure. We should consider creating a separate section on how to specify colors, since the logic is shared with `%style`.

Reply:
@ppkarwasz, would you mind pushing this improvement in a commit to this branch, please?

Reply:
Fixed in 8950932be4e2865d5fc2cddce84da2b4cb52e940, except for the `DISABLEANSI` and `NOCONSOLEANSI`, which seem more like an internal hack to pass the value of the `PatternLayout` configuration options to the converter, than real options.